linkerd/app/inbound/src/policy/http/tests.rs

```diff
@@ -130,6 +143,69 @@ async fn http_route() {
         .is::<HttpRouteNotFound>());
 }
 
+#[tokio::test(flavor = "current_thread")]
+async fn http_filter_header() {
+    use linkerd_server_policy::http::{filter, r#match::MatchRequest, Filter, Policy, Route, Rule};
+
+    let rmeta = Arc::new(Meta::Resource {
+        group: "gateway.networking.k8s.io".into(),
+        kind: "httproute".into(),
+        name: "testrt".into(),
+    });
+    let proto = Protocol::Http1(Arc::new([Route {
+        hosts: vec![],
+        rules: vec![Rule {
+            matches: vec![MatchRequest {
+                method: Some(::http::Method::GET),
+                ..MatchRequest::default()
+            }],
+            policy: Policy {
+                authorizations: Arc::new([Authorization {
+                    authentication: Authentication::Unauthenticated,
+                    networks: vec![std::net::IpAddr::from([192, 168, 3, 3]).into()],
+                    meta: Arc::new(Meta::Resource {
+                        group: "policy.linkerd.io".into(),
+                        kind: "server".into(),
+                        name: "testsaz".into(),
+                    }),
+                }]),
+                filters: vec![Filter::RequestHeaders(filter::ModifyRequestHeader {
+                    add: vec![("testkey".parse().unwrap(), "testval".parse().unwrap())],
+                    ..filter::ModifyRequestHeader::default()
```

Comment:
seems like it might be worth also having tests for `set` and `remove`, but since the actual filter mechanics are fairly simple and just call a `HeaderMap` method, that may not be worth the boilerplate?

Reply:
i don't think that needs to be tested from here. We could have tests for the `filter::ModifyHeader` type that exercised that, but this is just ensuring that the filter gets applied by the service. There's nothing that could change in this service to keep this test passing while breaking set/remove.